### ISSUE TITLE
Null events from metrics with no keys

### ISSUE THREAD

**Issue by suyograo** [elastic/logstash#6449]
```
"events" : {
              "filtered" : 0,
              "in" : null,
              "duration_in_millis" : null,
              "out" : 0
            },
```

When we query the metrics snapshot, using `::extract_metrics`, if the key does not exist, it gives a null. This may be fine for certain consumers, but when indexing to ES, this ain't great because of the strict mapping. One option: `extract_metrics(key, default_value)` so we can return that if key is not found.

**Comment by ph** [comment]
As a rule of thumb we should populate the metric everytime with zeroes values. 
The metric store is dynamic in nature and unless something is record he is not aware that the values can exist.

**Comment by suyograo** [comment]
@ph ++. Looks like it's just the fields `in` and `duration_in_millis` that have not been init'd for global event metrics

**Comment by ph** [comment]
@suyograo I will make a quick fix for 5.2,  theses stats are collected at the batch level and we will refactor that soon, it will be a good time to move to an observer pattern, since the metric collecting are implemented in two different queues. :(
